packages/rspack/src/plugins/utils/apply-web-config.ts

```diff
@@ -0,0 +1,459 @@
+import {
+  type RspackPluginInstance,
+  type Configuration,
+  type RuleSetRule,
+  LightningCssMinimizerRspackPlugin,
+  DefinePlugin,
+  HtmlRspackPlugin,
+  CssExtractRspackPlugin,
+  EnvironmentPlugin,
+} from '@rspack/core';
+import { instantiateScriptPlugins } from './instantiate-script-plugins';
+import path, { join } from 'path';
+import { SubresourceIntegrityPlugin } from 'webpack-subresource-integrity';
+import { getOutputHashFormat } from './hash-format';
+import { normalizeExtraEntryPoints } from './normalize-entry';
+import {
+  getCommonLoadersForCssModules,
+  getCommonLoadersForGlobalCss,
+  getCommonLoadersForGlobalStyle,
+} from './loaders/stylesheet-loaders';
+import { NormalizedNxAppRspackPluginOptions } from './models';
+
+export function applyWebConfig(
+  options: NormalizedNxAppRspackPluginOptions,
+  config: Configuration = {},
+  {
+    useNormalizedEntry,
+  }: {
+    // rspack.Configuration allows arrays to be set on a single entry
+    // rspack then normalizes them to { import: "..." } objects
+    // This option allows use to preserve existing composePlugins behavior where entry.main is an array.
+    useNormalizedEntry?: boolean;
+  } = {}
+): void {
+  if (!process.env['NX_TASK_TARGET_PROJECT']) return;
```

Comment:
I know this was copied from the webpack code, but I think it should be checking `global.NX_GRAPH_CREATION` instead.

Reply:
I can update webpack too in that case

Reply:
updated